lib/jnpr/junos/device.py

```diff
@@ -638,27 +641,72 @@ def execute(self, rpc_cmd, **kvargs):
     # facts
     # ------------------------------------------------------------------------
 
-    def facts_refresh(self, exception_on_failure=False):
+    def facts_refresh(self,
+                      exception_on_failure=False,
+                      warnings_on_failure=None,
+                      keys=None):
         """
-        Reload the facts from the Junos device into :attr:`facts` property.
-
-        :param bool exception_on_failure: To raise exception or warning when
-                             facts gathering errors out.
-
+        Refresh the facts from the Junos device into :attr:`facts` property.
+        See :module:`jnpr.junos.facts` for a complete list of available facts.
+        For old-style facts, this causes all facts to be immediately reloaded.
+        For new-style facts, the current fact value(s) are deleted, and the
+        fact is reloaded on demand.
+
+        :param bool exception_on_failure: To raise exception when facts
+                                          gathering errors out. If True when
+                                          new-style fact gathering is in use,
+                                          causes all facts to be reloaded
+                                          rather than being loaded on demand.
+        :param bool warnings_on_failure: To print a warning when fact gathering
+                                         errors out.
+                                         warnings_on_failure=True is the
+                                         default for old-style facts.
+                                         warnings_on_failure=False is the
+                                         default for new-style facts.
+                                         If True when new-style fact gathering
+                                         is in use, causes all facts to be
+                                         reloaded rather than being loaded on
+                                         demand.
+        :param str, set, list, or tuple keys: The set of keys in facts to
```

Comment:
Can we add an example for this option?
Any why we have this rule, "and exception_on_failure==False and warnings==False"